include/dlaf/eigensolver/gen_to_std/mc/gen_to_std_L.h

```diff
@@ -0,0 +1,121 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/include/threads.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/pipeline.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/executor.h"
+#include "dlaf/communication/functions_sync.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/tile_output.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Implementation based on LAPACK Algorithm for the transformation from generalized to standard
+// eigenproblem (xHEGST)
+
+// Local implementation
+template <class T>
+void genToStd_L(Matrix<T, Device::CPU>& mat_a, Matrix<T, Device::CPU>& mat_l) {
+  constexpr auto Right = blas::Side::Right;
+  constexpr auto Left = blas::Side::Left;
+  constexpr auto Lower = blas::Uplo::Lower;
+  constexpr auto NonUnit = blas::Diag::NonUnit;
+  constexpr auto NoTrans = blas::Op::NoTrans;
+  constexpr auto ConjTrans = blas::Op::ConjTrans;
+
+  using hpx::util::unwrapping;
+
+  using hpx::threads::executors::pool_executor;
+  using hpx::threads::thread_priority_high;
+  using hpx::threads::thread_priority_default;
+
+  // Set up executor on the default queue with high priority.
+  pool_executor executor_hp("default", thread_priority_high);
+
+  // Set up executor on the default queue with default priority.
+  pool_executor executor_normal("default", thread_priority_default);
+
+  SizeType m = mat_a.nrTiles().rows();
+  SizeType n = mat_a.nrTiles().cols();
+
+  for (SizeType k = 0; k < n; ++k) {
+    auto kk = LocalTileIndex{k, k};
+
+    // Direct transformation to standard eigenvalue problem of the diagonal tile
+    hpx::dataflow(executor_hp, unwrapping(tile::hegst<T, Device::CPU>), 1, Lower, std::move(mat_a(kk)),
+                  std::move(mat_l(kk)));
+
+    if (k != (n - 1)) {
+      for (SizeType i = k + 1; i < m; ++i) {
+        // Working on panel...
+        auto ik = LocalTileIndex{i, k};
+        hpx::dataflow(executor_normal, hpx::util::unwrapping(tile::trsm<T, Device::CPU>), Right, Lower,
```

Comment:
Thanks to the `using` statement at the beginning of the function, you can simply refer to `unwrapping` without fully qualifying it (here and in all other places in this function).